### PR TITLE
Changed getting started docs for .NET to coded instrumentation

### PR DESCRIPTION
**Pull request by martinjt**
## Docs PR Checklist

This reverts the docs to using the .NET coded instrumentation that is more intuitive for .NET developers. This was discussed in the OTel-dotnet SIG and agreed that this is the approach we want to push for as the initial getting started as it's what .NET engineers expect to see.

- [x] This PR is for a documentation page whose authoritative copy is in the
      opentelemetry.io repository.

---

**Preview**: https://deploy-preview-3364--opentelemetry.netlify.app/docs/instrumentation/net/getting-started/

### CODE REVIEW

File: content/en/docs/instrumentation/net/getting-started.md [hunk 1]
```diff
@@ -45,14 +45,7 @@ using System.Globalization;
...
 
-var logger = app.Logger;
-
-int RollDice()
-{
-    return Random.Shared.Next(1, 7);
-}
-
-string HandleRollDice(string? player)
+string HandleRollDice([FromServices]ILogger<Program> logger, string? player)
```
Comment (cijothomas) [inline content/en/docs/instrumentation/net/getting-started.md:48]: we still need the int RollDice() method,?

File: content/en/docs/instrumentation/net/getting-started.md [hunk 2]
```diff
@@ -103,97 +96,53 @@ dotnet run
...
-     OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED=true \
-     OTEL_DOTNET_AUTO_LOGS_CONSOLE_EXPORTER_ENABLED=true
-     OTEL_SERVICE_NAME=RollDiceService
-   . $HOME/.otel-dotnet-auto/instrument.sh
-   ```
+```csharp
+var builder = WebApplication.CreateBuilder(args);
 
-   {{% /tab %}} {{% tab PowerShell - Windows %}}
+builder.Logging.AddOpenTelemetry(options =>
```
Comment (cijothomas) [inline content/en/docs/instrumentation/net/getting-started.md:114]: attach resource to logging as well.

File: content/en/docs/instrumentation/net/getting-started.md [hunk 3]
```diff
@@ -103,97 +96,53 @@ dotnet run
...
-   $env:OTEL_TRACES_EXPORTER="none"
-   $env:OTEL_METRICS_EXPORTER="none"
-   $env:OTEL_LOGS_EXPORTER="none"
-   $env:OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED="true"
-   $env:OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED="true"
-   $env:OTEL_DOTNET_AUTO_LOGS_CONSOLE_EXPORTER_ENABLED="true"
-   Register-OpenTelemetryForCurrentSession -OTelServiceName "RollDiceService"
-   ```
+builder.Services.AddOpenTelemetry()
+       .ConfigureResource(rb => rb.AddService("roll-dice"))
```
Comment (cijothomas) [inline content/en/docs/instrumentation/net/getting-started.md:120]: nit: the example output below needs to be updated to "roll-dice"

File: content/en/docs/instrumentation/net/getting-started.md [hunk 4]
```diff
@@ -103,97 +96,53 @@ dotnet run
...
-   suitable for your shell/terminal environment &mdash; we illustrate a notation
-   for bash-like shells and PowerShell:
+```shell
+dotnet add package OpenTelemetry.Extensions.Hosting
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore --prerelease
+dotnet add package OpenTelemetry.Exporter.Console
+```
 
-   {{< tabpane text=true >}} {{% tab Unix-shell %}}
+2. Setup the packages
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:109]: Setup the packages where? I guess in `Program.cs` at the top?

File: content/en/docs/instrumentation/net/getting-started.md [hunk 5]
```diff
@@ -103,97 +96,53 @@ dotnet run
...
-   $env:OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED="true"
-   $env:OTEL_DOTNET_AUTO_LOGS_CONSOLE_EXPORTER_ENABLED="true"
-   Register-OpenTelemetryForCurrentSession -OTelServiceName "RollDiceService"
-   ```
+builder.Services.AddOpenTelemetry()
+       .ConfigureResource(rb => rb.AddService("roll-dice"))
+       .WithTracing(tpb => tpb
+          .AddAspNetCoreInstrumentation()
+          .AddConsoleExporter())
+       .WithMetrics(mpb => mpb
```
Comment (cijothomas) [inline content/en/docs/instrumentation/net/getting-started.md:124]: nit: the sample output shows 'process.runtime.dotnet.gc.collections.count' , which needs to be cleaned up too.

**Review by svrnm — CHANGES_REQUESTED** [review]
I am wondering if we can take the instructions for automatic instrumentation and move some of it to a page /net/automatic/_index ?

**Comment by cartermp** [comment]
I agree, we should move the autoinstrumentation docs to an autoinstrumentation folder. There will be people who would prefer to use it and we should still have that documented, even if it's not the default for getting started.

File: content/en/docs/instrumentation/net/getting-started.md
Comment (utpilla) [inline content/en/docs/instrumentation/net/getting-started.md:1]: Need to remove the part about auto instrumentation under the "Example Application" as well (L24):

> If you are not using ASP.NET Core, that's OK — you can still use
OpenTelemetry .NET Automatic Instrumentation.

File: content/en/docs/instrumentation/net/getting-started.md (hunk 2 above)
Comment (utpilla) [reply content/en/docs/instrumentation/net/getting-started.md:114]: `ConfigureResource` called on `builder.Services.AddOpenTelemetry()` should add the configured resource to the logger provider as well.

@CodeBlanch Could you please confirm?

File: content/en/docs/instrumentation/net/getting-started.md [hunk 6]
```diff
@@ -103,97 +96,53 @@ dotnet run
...
-   $env:OTEL_METRICS_EXPORTER="none"
-   $env:OTEL_LOGS_EXPORTER="none"
-   $env:OTEL_DOTNET_AUTO_TRACES_CONSOLE_EXPORTER_ENABLED="true"
-   $env:OTEL_DOTNET_AUTO_METRICS_CONSOLE_EXPORTER_ENABLED="true"
-   $env:OTEL_DOTNET_AUTO_LOGS_CONSOLE_EXPORTER_ENABLED="true"
-   Register-OpenTelemetryForCurrentSession -OTelServiceName "RollDiceService"
-   ```
+builder.Services.AddOpenTelemetry()
+       .ConfigureResource(rb => rb.AddService("roll-dice"))
+       .WithTracing(tpb => tpb
```
Comment (utpilla) [inline content/en/docs/instrumentation/net/getting-started.md:121]: Could you rename the lambda expression input parameters to more friendly and intuitive names? We recently updated the aspnetcore examples in the main repo to do that:
- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/trace/getting-started-aspnetcore/Program.cs#L24-L29
- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/logs/getting-started-aspnetcore/Program.cs#L29-L42
- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/docs/metrics/getting-started-aspnetcore/Program.cs#L23-L31

rb -> resource
tpb -> tracing
mpb -> metrics
options -> logging (For the logger provider configuration)

File: content/en/docs/instrumentation/net/getting-started.md
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:1]: Please update line 10 as well: `You will learn how you can instrument a simple .NET application automatically,`

File: content/en/docs/instrumentation/net/getting-started.md [hunk 7]
```diff
@@ -21,8 +21,7 @@ Ensure that you have the following installed locally:
 
 The following example uses a basic
 [Minimal API with ASP.NET Core](https://learn.microsoft.com/aspnet/core/tutorials/min-web-api)
-application. If you are not using ASP.NET Core, that's OK — you can still use
-OpenTelemetry .NET Automatic Instrumentation.
+application.
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:24]:
```suggestion
application. If you are not using a minimal API with ASP.NET Core, that's OK — you can use OpenTelemetry .NET with other frameworks as well. For a complete list of libraries for supported frameworks, see the [registry](https://deploy-preview-3364--opentelemetry.netlify.app/ecosystem/registry/?component=instrumentation&language=dotnet).
```
Similar to other language getting started guides, point out that other setups / frameworks should work as well and that they can take a look into the registry to find them

File: content/en/docs/instrumentation/net/getting-started.md [hunk 8]
```diff
@@ -103,97 +95,58 @@ dotnet run
 
 ## Instrumentation
 
-Next, you'll use a [OpenTelemetry .NET Automatic Instrumentation](../automatic)
-to instrument the application at launch time. While you can [configure .NET
-Automatic Instrumentation][] in a number of ways, the steps below use Unix-shell
-or PowerShell scripts.
-
-> **Note**: PowerShell commands require elevated (administrator) privileges.
+Next we'll install the instrumentation nuget packages from OpenTelemetry that will generate the telemetry, and set them up.
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:98]: Is there a website you can point to for the nuget packages?

Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:98]:
```suggestion
Next we'll install the instrumentation [NuGet packages from OpenTelemetry](https://www.nuget.org/profiles/OpenTelemetry) that will generate the telemetry, and set them up.
```

**Comment by svrnm** [comment]
@martinjt anything you need from us to push this forward? Happy to have this change implemented.

**Comment by martinjt** [comment]
Sorry, I'd completely missed that this wasn't finished. Looking now

**Comment by martinjt** [comment]
Pretty sure it's ready now.

**Comment by martinjt** [comment]
@svrnm do let me know if you need anything else, might be easier to ping me on slack though

File: content/en/docs/instrumentation/net/getting-started.md [hunk 9]
```diff
@@ -202,49 +159,49 @@ that looks something like this (output is line-wrapped for readability):
 <summary>Traces and Logs</summary>
 
 ```log
-LogRecord.Timestamp:               2023-08-14T06:44:53.9279186Z
-LogRecord.TraceId:                 3961d22b5f90bf7662ad4933318743fe
-LogRecord.SpanId:                  93d5fcea422ff0ac
+ogRecord.Timestamp:               2023-10-23T12:13:30.2704325Z
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:162]:
```suggestion
LogRecord.Timestamp:               2023-10-23T12:13:30.2704325Z
```

File: content/en/docs/instrumentation/net/getting-started.md [hunk 10]
```diff
@@ -202,49 +159,49 @@ that looks something like this (output is line-wrapped for readability):
...
 Activity.Tags:
-    net.host.name: localhost
-    net.host.port: 8080
+    net.host.name: 127.0.0.1
+    net.host.port: 5197
     http.method: GET
     http.scheme: http
     http.target: /rolldice
-    http.url: http://localhost:8080/rolldice
+    http.url: http://127.0.0.1:5197/rolldice
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:195]:
```suggestion
    http.url: http://127.0.0.1:8080/rolldice
```

Please make sure that the sample app runs on the same port as suggested above (8080 to be consistent across language docs)

File: content/en/docs/instrumentation/net/getting-started.md [hunk 11]
```diff
@@ -202,49 +159,49 @@ that looks something like this (output is line-wrapped for readability):
...
 Activity.Kind:               Server
-Activity.StartTime:          2023-08-14T06:44:53.9278162Z
-Activity.Duration:           00:00:00.0049754
+Activity.StartTime:          2023-10-23T12:13:30.2163005Z
+Activity.Duration:           00:00:00.0585187
 Activity.Tags:
-    net.host.name: localhost
-    net.host.port: 8080
+    net.host.name: 127.0.0.1
+    net.host.port: 5197
```
Comment (svrnm) [inline content/en/docs/instrumentation/net/getting-started.md:191]:
```suggestion
    net.host.port: 8080
```

**Review by svrnm — CHANGES_REQUESTED** [review]
A few minor changes, but overall lgtm. 

The piece missing for me is that the instructions for the automatic instrumentation are still something worth keeping, so maybe as part of https://opentelemetry.io/docs/instrumentation/net/automatic/ there should be a section auto instrumenting the Getting Started app. Would this be something you can add to this PR or should we move this out into a separate one?

File: content/en/docs/instrumentation/net/getting-started.md [hunk 12]
```diff
@@ -103,97 +97,60 @@ dotnet run
...
-
-3. Set and export variables that specify a [console exporter][], then execute
-   script configuring other necessary environment variables using a notation
-   suitable for your shell/terminal environment &mdash; we illustrate a notation
-   for bash-like shells and PowerShell:
+```shell
+dotnet add package OpenTelemetry.Extensions.Hosting
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore --prerelease
+dotnet add package OpenTelemetry.Exporter.Console
+```
```
Comment (chalin) [inline content/en/docs/instrumentation/net/getting-started.md:110]: If you look at the preview, you'll see that the list item numbering is off: https://deploy-preview-3364--opentelemetry.netlify.app/docs/instrumentation/net/getting-started/#instrumentation

You need to indent the list-item contents, like this:

~~~suggestion
    ```shell
    dotnet add package OpenTelemetry.Extensions.Hosting
    dotnet add package OpenTelemetry.Instrumentation.AspNetCore --prerelease
    dotnet add package OpenTelemetry.Exporter.Console
    ```
~~~

If you indent all list item content correctly, then Prettier will sequentially number all list items.

**Review by chalin — CHANGES_REQUESTED** [review]

**Comment by cartermp** [comment]
@svrnm let's keep this one smaller and make additional notes w.r.t. automatic instrumentation in a different PR.

**Review by svrnm — APPROVED** [review]
Docs-wise this looks good, @open-telemetry/dotnet-approvers PTAL!

**Comment by pellared** [comment]
Assuming that this is merged: who (and when) wants to add getting started docs for automatic instrumentation?

**Comment by svrnm** [comment]
> Assuming that this is merged: who (and when) wants to add getting started docs for automatic instrumentation?

If nobody is going to take it, I will raise an issue and try to find some time to make a copy of what we have for automatic today

**Review by pellared — APPROVED** [review]

**Review by chalin — APPROVED** [review]
LGTM

**Comment by svrnm** [comment]
🎉

**Pull request merged**